icepyx/core/variables.py

```diff
@@ -638,3 +657,19 @@ def remove(self, all=False, var_list=None, beam_list=None, keyword_list=None):
                             del self.wanted[vkey]
                     except KeyError:
                         pass
+
+    # DevNote: This is a modified function from the Query class. 
+    def _get_latest_version(self):
```

Comment:
Should we move a version of this function to is2ref instead of duplicating it here and in query? We could allow passing in `product` or `about_info` to accommodate the Variables and Query cases, respectively. Or is this a case where the function itself isn't long enough to warrant not duplicating it?

Reply:
I think it's a good idea to move it into is2ref. I was unsure at first, because the function currently sets the `_about_product` instance attribute when it gets run. If we move `latest_version` to is2ref we can't set that attribute anymore. Looking through query a second time, though, I think it's fine to move `latest_version` is2ref. `latest_version()` only gets called in init, so we don't have to worry about someone calling it multiple times. In theory we loose a little efficiency the next time someone calls `product_summary_info` or `product_all_info`, but my two cents is that speed is worth it to not have a function duplicated.

I just pushed a commit with the function moved, so feel free to comment further in review.

Reply:
> In theory we loose a little efficiency the next time someone calls product_summary_info or product_all_info, but my two cents is that speed is worth it to not have a function duplicated.

Agreed. If only we knew how frequently these functions are used...